jira/client.py

```diff
@@ -2422,6 +2424,95 @@ def kill_session(self):
         url = self._options['server'] + '/rest/auth/latest/session'
         return self._session.delete(url)
 
+    # Webhook
+
+    def webhook(self, id):
+        """Get a webhook Resource from the server.
+
+        :param id: ID of the webhook to get
+        """
+        webhook = Webhook(self._options, self._session)
+        webhook.find(id)
+        return webhook
+
+    def create_webhook(self, fields=None, prefetch=True, **fieldargs):
+        """Create a new webhook and return an webhook Resource for it.
+
+        Each keyword argument (other than the predefined ones) is treated as a
+        field name and the argument's value is treated as the intended value for
+        that field -- if the fields argument is used, all other keyword arguments
+        will be ignored.
+
+        By default, the client will immediately reload the webhook Resource
+        created by this method in order to return a complete Webhook object to
+        the caller; this behavior can be controlled through the 'prefetch' argument.
+
+        JIRA projects may contain many different webhook types. Some webhook
+        screens have different requirements for fields in a new webhook. This
+        information is available through the 'createmeta' method. Further examples are
+        available here:
+        https://developer.atlassian.com/server/jira/platform/webhooks/
+
+        :param fields: a dict containing field names and the values to use.
+                       If present, all other keyword arguments will be ignored
+        :param prefetch: whether to reload the created webhook Resource so that
+                         all of its data is present in the value returned from this method
+        """
+        data = {}
+        if fields is not None:
+            data['fields'] = fields
+        else:
+            fields_dict = {}
+            for field in fieldargs:
+                fields_dict[field] = fieldargs[field]
+            data['fields'] = fields_dict
+
+        url = self._get_url('webhook', base=self.WEBHOOK_BASE_URL)
+        json_data = json.dumps(fieldargs)
+        try:
+            r = self._session.post(url, data=json_data)
```

Comment:
You should be able to specify `json=fieldargs` instead of needing to explicitly `json.dump` it first..

Reply:
@dbaxa can you provide an example of what you mean I think I may be missing something.

Reply:
@dbaxa Are you saying just forego the `json_data` and just replace with `fieldargs`